static/js/apps/eval_retrieval_generation/query_section.tsx

```diff
@@ -171,14 +174,8 @@ export function QuerySection(): JSX.Element {
       feedbackStage,
       sessionQueryId
     ).then(({ answer, metadata }) => {
-      // Set empty answer if metadata doesn't match current state.
-      if (
-        metadata.evalType !== evalType ||
-        metadata.feedbackStage !== feedbackStage ||
-        metadata.sessionQueryId !== sessionQueryId
-      ) {
-        return;
-      } else {
+      // Only set answer if it matches the current answer metadata
+      if (_.isEqual(answerMetadata.current, metadata)) {
```

Comment:
in which case the previous failed?

Reply:
this useEffect listens on both queryId and feedbackStage & when we move on to the next query, both those dependencies will change one after the other & depending on which promise finishes last, that's the answer that will show up